File: src/AcceptanceSpecSystem.Data/Entities/Keyword.cs
namespace AcceptanceSpecSystem.Data.Entities;

/// <summary>
/// 关键字实体
/// </summary>
public class Keyword
{
    /// <summary>
    /// 关键字ID
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// 关键字内容（唯一）
    /// </summary>
    public string Word { get; set; } = string.Empty;

    /// <summary>
    /// 创建时间
    /// </summary>
    public DateTime CreatedAt { get; set; } = DateTime.Now;
}

File: web/src/api/keyword.ts
import { http } from "@/utils/http";
import type { ApiResponse, PagedData, PagedRequest } from "./customer";

export interface Keyword {
  id: number;
  word: string;
  createdAt: string;
}

export interface KeywordListRequest extends PagedRequest {}

export interface CreateKeywordRequest {
  word: string;
}

export interface UpdateKeywordRequest {
  word: string;
}

export interface BatchAddKeywordsRequest {
  words: string[];
}

const baseUrl = "/api/keywords";

export const getKeywordList = (params?: KeywordListRequest) => {
  return http.request<ApiResponse<PagedData<Keyword>>>("get", baseUrl, {
    params
  });
};

export const createKeyword = (data: CreateKeywordRequest) => {
  return http.request<ApiResponse<Keyword>>("post", baseUrl, { data });
};

export const updateKeyword = (id: number, data: UpdateKeywordRequest) => {
  return http.request<ApiResponse<void>>("put", `${baseUrl}/${id}`, { data });
};

export const deleteKeyword = (id: number) => {
  return http.request<ApiResponse<void>>("delete", `${baseUrl}/${id}`);
};

export const batchAddKeywords = (data: BatchAddKeywordsRequest) => {
  return http.request<ApiResponse<number>>("post", `${baseUrl}/batch`, { data });
};

